CMakeLists.txt

```diff
@@ -6,7 +6,9 @@ if (CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
   set(nfd_ROOT_PROJECT ON)
 endif ()
 
-option(BUILD_SHARED_LIBS "Build a shared library instead of static" OFF)
+if(NOT DEFINED BUILD_SHARED_LIBS)
+	option(BUILD_SHARED_LIBS "Build a shared library instead of static" OFF)
+endif()
```

Comment:
Can you explain a bit more about what this does?  Most examples on the Internet do not put `option` inside a if condition like this.  Some GitHub Actions are configured to pass `-DBUILD_SHARED_LIBS=ON`, and it doesn't cause any problems with the existing code.

Also, what is making `BUILD_SHARED_LIBS` different from `NFD_BUILD_TESTS` and `NFD_INSTALL` below?

Reply:
Yes, I am not building NFDE by itself.

Here is the warning I'm getting:
```
CMake Warning (dev) at deps/nfde/CMakeLists.txt:9 (option):
  Policy CMP0077 is not set: option() honors normal variables.  Run "cmake
  --help-policy CMP0077" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

  For compatibility with older versions of CMake, option is clearing the
  normal variable 'BUILD_SHARED_LIBS'.
This warning is for project developers.  Use -Wno-dev to suppress it.
```

BUILD_SHARED_LIBS is a standard CMake variable used by a lot of different projects, though it might make sense to do that with the others as well for projects like mine that bundle NFDE.